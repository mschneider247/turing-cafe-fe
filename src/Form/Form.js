import React, { Component } from 'react';
import './Form.css'

class Form extends Component {
  constructor() {
    super()
    this.state = {
      id: Date.now(),
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  updateFormState = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <section>
        <input
          type='text' 
          placeholder='Name'
          name='name'
          value={this.state.name} 
          onChange={event => this.updateFormState(event)}>
        </input>
        <input
          type='text'  
          placeholder='Date (mm/dd)'
          name='date'
          value={this.state.date} 
          onChange={event => this.updateFormState(event)}>
        </input>
        <input
          type='text'  
          placeholder='Time' 
          name='time'
          value={this.state.time}
          onChange={event => this.updateFormState(event)}>
        </input>
        <input
          type='number'  
          placeholder='Number of Guests'
          name='number'
          value={this.state.guests} 
          onChange={event => this.updateFormState(event)}>
        </input>
        <button onClick={() => this.props.addReservation(this.state)}>Make Reservation</button>
      </section>
    )
  }
}

export default Form