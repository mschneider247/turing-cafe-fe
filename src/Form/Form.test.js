import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {

  it('Should match the snapshot', () => {
    const wrapper = shallow(<Form />)
    expect(wrapper).toMatchSnapshot();
  })
})