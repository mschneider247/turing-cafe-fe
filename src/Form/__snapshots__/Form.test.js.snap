// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match the snapshot 1`] = `
<section>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</section>
`;
